src/App.vue

```diff
@@ -6,69 +6,148 @@
       </div>
 
       <div v-for="tile in tiles"
-            v-bind:id="tile.id"
+            :key="tile.id"
             v-bind:style="{top: `${ tile.y * 60}px`, left: `${ tile.x * 60}px`}"
             class="tile">
         {{tile.number}}
       </div>
+
+      <div v-show="score" class="game-over-shroud">
+        <div class="score">
+          Your score is {{score}}
+        </div>
+        <div class="game-over-text">
+          Game Over
+        </div>
+        <div v-on:click="newGame" class="new-game-shroud-button">
+          New Game
+        </div>
+      </div>
+    </div>
+    <div v-on:click="newGame" class="new-game-button">
+      New Game
     </div>
   </div>
 </template>
 
 <script>
 
-const transitionInterval = 0
+const transitionInterval = 400
 
 export default {
   name: 'app',
   created: function () {
+    this.newGame()
     window.addEventListener('keydown', this.move)
   },
   data: function () {
     return {
-      tiles: [
-               {x: 1, y: 2, number: 1},
-               {x: 2, y: 2, number: 2},
-               {x: 3, y: 2, number: 3}
-             ]
+      tiles: [],
+      score: 0,
+      locked: false
     }
   },
   methods: {
+    newGame() {
+      this.initializeTiles()
+      this.score = 0
+    },
+
+    initializeTiles() {
+      this.tiles = []
+
+      let tileNumbers = [1, 1, 2, 2, 3, 3]
+
+      for(let i = 0; i < 3; i++) {
+        tileNumbers.push(Math.floor(Math.random() * 3 + 1))
+      }
+
+      tileNumbers.forEach(number => {
+        let x, y
+
+        do {
+          x = Math.floor(Math.random() * 4 + 1)
+          y = Math.floor(Math.random() * 4 + 1)
+        } while (this.tiles.find(tile => tile.x === x && tile.y === y))
+
+        this.tiles.push({x, y, number, id: Math.random()})
+      })
+    },
+
+    gameOverCheck () {
+      const availableCols = []
+      const availableRows = []
+
+      for (let colNum = 1; colNum <= 4; colNum++) {
+        if (this.willThereBeSpaceInCol(colNum)) {
+          availableCols.push(colNum)
+        }
+      }
+
+      for (let rowNum = 1; rowNum <= 4; rowNum++) {
+        if (this.willThereBeSpaceInRow(rowNum)) {
+          availableRows.push(rowNum)
+        }
+      }
+
+      if (availableRows.length !== 0 || availableCols.length !== 0) return
```

Comment:
this function is called `gameOverCheck`, which makes me think it should return a `boolean`. But looks like it is returning `undefined` and the score?